addons/web/static/src/webclient/actions/action_service.js

```diff
@@ -1110,6 +1092,24 @@ export function makeActionManager(env, router = _router) {
         };
         action.controllers[view.type] = controller;
 
+        if (options.newStack?.at(-1)?.lazyController) {
+            const multiView = action.views.find(
+                (view) => view[1] !== "form" && view[1] !== "search"
+            );
+            if (multiView) {
+                // If the current action has a multi-record view, we add the last
+                // controller to the breadcrumb controllers.
+                delete options.newStack.at(-1).lazyController;
+                options.newStack.at(-1).displayName = action.display_name || action.name || "";
+                options.newStack.at(-1).action = action;
+                options.newStack.at(-1).props.type = multiView[1];
+            } else {
+                // If the current action doesn't have a multi-record view,
+                // we don't need to add the last controller to the breadcrumb controllers
+                options.newStack.splice(-1);
```

Comment:
you remove the last one? Isn't it the form controller?

Reply:
Nop, the form controller is not in the newStack, it will add to the controllerStack with the newStack on updateUI

Reply:
What if the stack contains a lot of controller, and the server action is not the last one? It will be correctly handled by loadBreadcrumbs then?

Reply:
Yes, loadBreadcrumb will run the server action, and all the information (display_name, and action id) will be correctly put in the controllerStack